### PR TITLE
s1_reader.py: implement load_bursts speedups

### PR DESCRIPTION
**Pull request by scottstanie**
I added three speedups which make loading 1 subswath run about 3x faster:

1. use fromisoformat instead of strptime (https://stackoverflow.com/a/61710371/4174466)

2. use lxml to parse if available (https://stackoverflow.com/a/18507562/4174466)

3. use eval poly1d by passing array instead of a comprehension

I was checking on this with the following little program to load one subswath's bursts:
```python
(mapping) [staniewi@aurora hawaii]$ cat tests1.py
import s1reader
import time

tstart = time.time()
b = s1reader.load_bursts("data/S1B_IW_SLC__1SDV_20200927T161542_20200927T161609_023563_02CC58_C487.zip", "orbits/S1B_OPER_AUX_POEORB_OPOD_20210317T172346_V20200926T225942_20200928T005942.EOF", 2)
```

First, with the current code:
```
$ python tests1.py
Elapsed: 775.102 ms
```

With change (1), the fromisoformat

```
$ python tests1.py
Elapsed: 577.467 ms
```

With change (2), the lxml parser
```
$ python tests1.py
Elapsed: 319.141 ms
```

With (3), with the Poly1D evaluation change using isce3 instead of the list comprehension:

```
$ python tests1.py
Elapsed: 257.990 ms
```

### CODE REVIEW

**Comment by scottstanie** [comment]
Just a note on using lxml optionally- although it is quite common, it does involve a separate install. So the question is whether we think the 1.5x - 2x speedup is worth adding that in.

**Comment by LiangJYu** [comment]
> Just a note on using lxml optionally- although it is quite common, it does involve a separate install. So the question is whether we think the 1.5x - 2x speedup is worth adding that in.

I'm ok with having `lxml` as optional. It's a nice speed if anyone who only wants to read a bunch of SAFEs for whatever reason. I'm guessing in the context of running a workflow the improvement in processing it is not as significant?

**Comment by scottstanie** [comment]
> I'm guessing in the context of running a workflow the improvement in processing is not as significant?

Yes I believe that it's small for a workflow of one burst in COMPASS. The this PR came about was for the work i'm doing on the `s1_geo_stack.py`  script here: https://github.com/opera-adt/COMPASS/pull/53 . For a large stack, it took about 10 minutes to run, and 99% of the time was in `load_bursts`.

**Comment by yunjunz** [comment]
`lxml` is light and very commonly used. If it speeds up our code, it's perfectly fine with me to add it to the `requirements.txt` file.

**Comment by scottstanie** [comment]
Alright I took out the `try/except` in favor of adding it to the requirements file.

**Review by vbrancat — APPROVED** [review]
LGTM, thanks for this nice improvement

File: .gitignore [hunk 1]
```diff
@@ -116,6 +116,9 @@ venv.bak/
 .spyderproject
```
[the remark this answers is not in the document]
Comment (scottstanie) [reply .gitignore:116]: These are ignoring the config directories created by [spyder](https://www.spyder-ide.org/) and VSCode (I assume most came from something @LiangJYu found online based on how many there were in the first commit https://github.com/opera-adt/s1-reader/pull/1/files). Are you asking to add in .vscode directory?

**Comment by scottstanie** [comment]
I've updated the circleci files for `lxml`, but I now see that the Codacy issue is 

> Using lxml.etree to parse untrusted XML data is known to be vulnerable to XML attacks. Replace lxml.etree with the equivalent defusedxml package. 

Do we want to follow that and swap lxml for [defusedxml](https://github.com/tiran/defusedxml)? Or do we think that it's alright because the only XML's we're parsing are from Sentinel-1, rather than accepting arbitrary XML?

**Comment by yunjunz** [comment]
> Do we want to follow that and swap lxml for [defusedxml](https://github.com/tiran/defusedxml)? Or do we think that it's alright because the only XML's we're parsing are from Sentinel-1, rather than accepting arbitrary XML?

Yeah, I think we are parsing S1 XML only, I think we are fine using `lxml`, without `defusedxml`.

**Pull request merged**